### PR TITLE
Upgrade prometheus pushgateway charts to latest.

### PR DESCRIPTION
**Pull request by rakesh-garimella**
**Description**

Changes proposed in this pull request:

- Update prometheus pushgateway to latest upstream charts
- Carried over the way default label handling from the upstream chart:
```
metadata:
  labels:
{{ template "prometheus-pushgateway.defaultLabels" merge (dict "extraLabels" .Values.someAdditionalLabels) .  }}
```

**Related issue(s)**

https://github.com/kyma-project/kyma/issues/9897

### CODE REVIEW

**Review by suleymanakbas91 — CHANGES_REQUESTED** [review]

File: resources/monitoring/charts/prometheus-pushgateway/Chart.yaml [hunk 1]
```diff
@@ -1,7 +1,8 @@
 apiVersion: v1
+appVersion: "1.3.0"
```
[the remark this answers is not in the document]
Comment (rakesh-garimella) [reply resources/monitoring/charts/prometheus-pushgateway/Chart.yaml]: removed

File: resources/monitoring/charts/prometheus-pushgateway/templates/_helpers.tpl [hunk 2]
```diff
@@ -1,3 +1,8 @@
+
```
[the remark this answers is not in the document]
Comment (rakesh-garimella) [reply resources/monitoring/charts/prometheus-pushgateway/templates/_helpers.tpl]: fixed

**Review by suleymanakbas91 — APPROVED** [review]

**Pull request merged**